check_license_source_files.sh

```diff
@@ -188,10 +188,9 @@ EOF
         cat "$license" >&2
         TEST_RESULT=1
     else
-        copyright_modified=$(echo "${CM} Copyright <copyright_year> Northern.tech AS" | sed "s/<copyright_year>/$modified_year/g")
-        copyright_file="$(head -n 1 "$file")"
-        if [ "$copyright_modified" != "$copyright_file" ]; then
-            echo >&2 "!!! FAILED license check on ${orig_file}; make sure copyright year matches last modified year of the file ($modified_year)"
+        copyright_year=$(grep -oP "${CM} Copyright \K\d{4} Northern.tech AS")
```

Comment:
Nothing is grepped here, `$file` is missing.

Also, I don't think `\K` will work since it includes the suffix portion, IOW the company name.

Reply:
Oh sorry, I didn't mean you could just remove it. I can't immediately see any way to do this with one grep alone, so how about:

```bash
        copyright_year=$(grep -oP "${CM} Copyright \d{4} Northern.tech AS" "$file" | grep -oP "\d{4}")
```

Reply:
ah grep seems slightly tricky to rely on 🙈 - Mac, "regular" grep and the alpine/ busybox grep all with their own surprises 😅... do you think `grep -oE` would work for you as well?

Reply:
I think `-E` should work. I believe `-P` was only necessary for `\K`.

Reply:
😞 unfortunately `-E` is not available in the `golang`/ `debian` based grep (but it is on mac + alpine)... so now I'm installing `grep` in the license checking stage instead and at least locally that works 🤞